reactor-core/src/main/java/reactor/core/publisher/FluxCreate.java

```diff
@@ -171,7 +172,7 @@ public Context currentContext() {
 		@Override
 		public void error(Throwable t) {
 			Objects.requireNonNull(t, "t is null in sink.error(t)");
-			if (sink.isDisposed() || done) {
+			if (sink.isTerminated() || done) {
```

Comment:
I wonder if we still need done?

Reply:
yes, for the same reason we still need it in `BufferAsyncSink`: swapping the base sink's `disposable` to `TERMINATED` instead of deferring until the drain loop calls `super.complete()`/`super.error(Throwable)` would mean either:
 - we'd also execute the current `disposable.dispose()` method => ⚠️ would be too early for draining variants
 - we'd have lost the original `disposable` handler by the time we call `super.complete()/error()`, so it never gets executed.